File: App10_ListaTarefas/app/src/main/java/com/cursoandroid/app10_listatarefas/MainActivity.java
package com.cursoandroid.app10_listatarefas;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;

import com.google.android.material.floatingactionbutton.FloatingActionButton;

public class MainActivity extends AppCompatActivity {

    FloatingActionButton fob;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        findA();

        fob.setOnClickListener(view -> {
            Intent intent = new Intent(MainActivity.this, AdicionarActivity.class);
            startActivity(intent);
        });
    }

    private void findA() {
        fob = findViewById(R.id.floatingActionButton);
    }
}

File: Barbearia/app/src/main/java/com/cursoandroid/barbearia/MainActivity.kt
package com.cursoandroid.barbearia

import android.content.Intent
import android.graphics.Color
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import com.cursoandroid.barbearia.databinding.ActivityMainBinding
import com.cursoandroid.barbearia.view.HomeActivity
import com.google.android.material.snackbar.Snackbar

class MainActivity : AppCompatActivity() {

    private lateinit var binding : ActivityMainBinding


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        supportActionBar?.hide()

        binding.button.setOnClickListener{
            val nome = binding.etNome.text.toString()
            val senha = binding.etSenha.text.toString()

            when{
                nome.isEmpty()->{
                    mensagem(it, "Digite seu nome")
                }senha.isEmpty()->{
                    mensagem(it, "Digite a senha")
                }senha.length<=5 ->{
                    mensagem(it, "A senha precisa ter pelo menos 6 caracteres")
                }else ->{
                    navegarParaHome(nome)
                }
            }
        }
    }

    private fun mensagem(view: View, mensagem: String){
        val snackbar= Snackbar.make(view, mensagem, Snackbar.LENGTH_SHORT)
        snackbar.setBackgroundTint(Color.parseColor("#FF0000"))
        snackbar.setTextColor(Color.parseColor("#FFFFFF"))
        snackbar.show()
    }

    private fun navegarParaHome (nome: String){
        val intent = Intent(this, HomeActivity::class.java)
        intent.putExtra("nome", nome)
        startActivity(intent)
    }
}